arangod/Replication2/Supervision/CollectionGroupSupervision.cpp

```diff
@@ -197,30 +221,34 @@ auto createCollectionGroupTarget(
   std::unordered_map<CollectionID, ag::CollectionPlanSpecification> collections;
   for (auto const& [cid, collection] : group.target.collections) {
     std::vector<ShardID> shardList;
-    std::generate_n(std::back_inserter(shardList),
-                    attributes.immutableAttributes.numberOfShards, [&] {
-                      return basics::StringUtils::concatT("s", uniqid.next());
-                    });
-
-    ADB_PROD_ASSERT(group.targetCollections.contains(cid))
-        << "collection " << cid << " is listed in collection group "
-        << group.target.id << " but not in Target/Collections";
     PlanShardToServerMapping mapping{};
-    for (size_t k = 0; k < shardList.size(); ++k) {
-      ResponsibleServerList serverids{};
-      auto const& log = replicatedLogs[spec.shardSheaves[k].replicatedLog];
-      serverids.servers.emplace_back(log.leader.value());
-      for (auto const& p : log.participants) {
-        if (p.first != log.leader) {
-          serverids.servers.emplace_back(p.first);
+    auto const& targetCollection = group.targetCollections.at(cid);
+
+    // If we have shadow collections we do not have any shards
+    if (!targetCollection.immutableProperties.shadowCollections.has_value()) {
+      std::generate_n(std::back_inserter(shardList),
+                      attributes.immutableAttributes.numberOfShards, [&] {
+                        return basics::StringUtils::concatT("s", uniqid.next());
+                      });
+
+      ADB_PROD_ASSERT(group.targetCollections.contains(cid))
+          << "collection " << cid << " is listed in collection group "
+          << group.target.id << " but not in Target/Collections";
+      for (size_t k = 0; k < shardList.size(); ++k) {
+        ResponsibleServerList serverids{};
+        auto const& log = replicatedLogs[spec.shardSheaves[k].replicatedLog];
+        serverids.servers.emplace_back(log.leader.value());
+        for (auto const& p : log.participants) {
+          if (p.first != log.leader) {
```

Comment:
A note for myself.: if the log's leader in Target is failed, the supervision will elect a new one, but this one remains set in Target.